contrib/hadoop-store-builder/src/java/voldemort/store/readonly/fetcher/HdfsCopyStats.java

```diff
@@ -161,6 +175,41 @@ public void recordBytesWritten(long bytesWritten) {
     public void recordBytesTransferred(long bytesTransferred) {
         this.totalBytesTransferred += bytesTransferred;
         this.bytesTransferredSinceLastReport += bytesTransferred;
+        // Update the aggregated stats for total bytes transferred
+        HdfsFetcherAggStats.getStats().recordBytesTransferred(bytesTransferred);
+    }
+
+    public static void storeFetch() {
+        HdfsFetcherAggStats.getStats().storeFetch();
+    }
+    public void singleFileFetchStart(boolean isRetry) {
+        HdfsFetcherAggStats.getStats().singleFileFetchStart(isRetry);
+    }
+
+    public void singleFileFetchEnd() {
+        HdfsFetcherAggStats.getStats().singleFileFetchEnd();
+    }
+
+    public void checkSumFailed() {
+        HdfsFetcherAggStats.getStats().checkSumFailed();
+    }
+
+    public static void reportExceptionForStats(Exception e) {
+        if (ExceptionUtils.recursiveClassEquals(e, AuthenticationException.class)) {
+            HdfsFetcherAggStats.getStats().authenticateFailed();
+        } else if (e instanceof QuotaExceededException) {
+            HdfsFetcherAggStats.getStats().quotaCheckFailed();
+        } else if (e instanceof UnauthorizedStoreException) {
+            HdfsFetcherAggStats.getStats().unauthorizedStorePush();
+        } else if (e instanceof FileNotFoundException) {
+            HdfsFetcherAggStats.getStats().fileNotFound();
+        } else {
+            HdfsFetcherAggStats.getStats().fileReadFailed();
```

Comment:
This should probably tracked as an "unknown" file fetch failure, probably best stored in unknownFileFetchFailures (or similarly named).